src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+package org.broadinstitute.hellbender.utils.config;
+
+import org.aeonbits.owner.Config;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.TestProgramGroup;
+import org.broadinstitute.hellbender.engine.GATKTool;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.ClassUtils;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Properties;
+
+/**
+ * Integration test
+ * Created by jonn on 12/7/17.
+ */
+public class ConfigIntegrationTest extends CommandLineProgramTest {
+
+    @CommandLineProgramProperties(
+            summary = "Dummy empty command line that requires a reference .",
+            oneLineSummary = "empty class",
+            programGroup = TestProgramGroup.class
+    )
+    public static class DummyGatkTool extends GATKTool {
```

Comment:
Make this dummy tool `private` (should still be runnable by the integration test infrastructure)

Reply:
@droazen I tried that first and I got an error saying that `DummyGatkTool` was not a valid tool name.

It seems like it will not work if they are private.  @lbergelson helped me with that one.